### PR TITLE
Choices in `SelectField` objects now handle `HTMLString` correctly.

### PR DESCRIPTION
**Pull request by sonovice**
Proposal for #301

### CODE REVIEW

**Comment by davidism** [comment]
You have mistakenly committed your IDE and build files.  Please strip those commits and force push.

**Comment by sonovice** [comment]
Thanks, I overlooked it.

**Comment by davidism** [comment]
Please add another test to explicitly test `escape_html` in addition to testing it indirectly.

File: wtforms/widgets/core.py [hunk 1]
```diff
@@ -14,6 +14,24 @@
...
+
+    If the optional flag quote is true (the default), the quotation mark
+    characters, both double quote (") and single quote (') characters are also
+    translated.
+
+    If a `HTMLString` is provied, it's assumed that whatever you give to
+    escape_html is a string with any unsafe values already escaped.
+    """
+    if hasattr(s, '__html__'):
+        s = text_type(s.__html__())
```
Comment (davidism) [inline wtforms/widgets/core.py]: Remove `text_type`, as this will unmark a string that is already marked safe.

**Pull request merged**